src/software/thunderscope/robot_diagnostics/robot_info.py

```diff
@@ -0,0 +1,191 @@
+        self.battery_progress_bar = common_widgets.ColorProgressBar(
+            MIN_BATTERY_VOLTAGE - 3, MAX_BATTERY_VOLTAGE
+        )
+        # Battery Voltage Label
+        self.battery_label = QLabel()
+        self.battery_label.setText("NAN")
+        # Label changes when voltage bar level changes
+        self.battery_progress_bar.floatValueChanged.connect(
+            lambda float_val: self.battery_label.setText("%.2fV" % float_val)
+        )
+        self.battery_layout.addWidget(self.battery_progress_bar)
+        self.battery_layout.addWidget(self.battery_label)
+
+        self.status_layout.addLayout(self.battery_layout)
+
+        # Control mode dropdown
+        self.control_mode_layout = QHBoxLayout()
+        self.control_mode_menu = self.create_control_mode_menu(load_fullsystem)
+
+        # Breakbeam status
+        self.breakbeam_label = QLabel()
+        self.breakbeam_label.setText("BREAKBEAM")
+        self.breakbeam_label.setStyleSheet("background-color: grey")
+        self.control_mode_layout.addWidget(self.breakbeam_label)
+        self.control_mode_layout.addWidget(self.control_mode_menu)
+
+        self.status_layout.addLayout(self.control_mode_layout)
+
+        # Vision Pattern
+        self.layout.addWidget(self.create_vision_pattern_label("b", 25))
+
+        self.layout.addLayout(self.status_layout)
+
+        self.setLayout(self.layout)
+
+    def create_control_mode_menu(self, load_fullsystem):
+        """
+        Creates the drop down menu to select the input for each robot
+        :param robot_id: the id of the robot this menu belongs to
+        :param load_fullsystem: whether fullsystem is also an option
+        :return: QComboBox object
+        """
+        control_mode_menu = QComboBox()
+
+        control_mode_menu.addItems(["None", "Manual"])
+        control_mode_menu.setCurrentIndex(0)
+
+        if load_fullsystem:
+            control_mode_menu.addItem("AI")
+            control_mode_menu.setCurrentIndex(2)
+
+        control_mode_menu.currentIndexChanged.connect(
+            lambda mode, robot_id=self.robot_id: self.toggle_one_connection_signal.emit(
+                mode, robot_id
+            )
+        )
+
+        return control_mode_menu
+
+    def create_vision_pattern_label(self, team_colour, radius):
+        """Given a robot id, team color and radius, draw the vision
+        pattern on a label and return it.
+
+        :param team_colour: The team colour
+        :param radius: The radius of the robot
+
+        """
+        pixmap = QtGui.QPixmap(radius * 2, radius * 2)
+        pixmap.fill(QtCore.Qt.GlobalColor.transparent)
+
+        painter = QtGui.QPainter(pixmap)
+        painter.setPen(pg.mkPen("black"))
+        painter.setBrush(pg.mkBrush("black"))
+
+        convert_degree = -16
+
+        painter.drawChord(
+            QtCore.QRectF(0, 0, int(radius * 2), int(radius * 2),),
+            -45 * convert_degree,
+            270 * convert_degree,
+        )
```

Comment:
You can probably refactor this slightly to have configurable radius and use it instead (might want to move it to common widgets or something):
https://github.com/UBC-Thunderbots/Software/blob/3ba4fb1671136091c9af5a9553a22495a27d704c/src/software/thunderscope/field/field_layer.py#L58-L70

Reply:
I added a method to common_widgets, but I had to pass in the bounding rectangle as a param. This is as field layer makes it's own rectangle with some offsets, and I thought it'd be messy to pass in x, y, width, and height to the function